sdk/containerregistry/container-registry/samples/v1-beta/typescript/src/deleteBlob.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary Deletes the blobs associated with a given manifest from the repository.
+ */
+
+import {
+  ContainerRegistryBlobClient,
+  isDownloadOciImageManifestResult,
+} from "@azure/container-registry";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  // Get the service endpoint from the environment
+  const endpoint = process.env.CONTAINER_REGISTRY_ENDPOINT || "<endpoint>";
+  const repository = process.env.CONTAINER_REGISTRY_REPOSITORY || "library/hello-world";
+  // Create a new ContainerRegistryClient
+  const client = new ContainerRegistryBlobClient(
+    endpoint,
+    repository,
+    new DefaultAzureCredential()
+  );
+
+  const downloadResult = await client.downloadManifest("latest");
+
+  if (!isDownloadOciImageManifestResult(downloadResult)) {
```

Comment:
Do we use the same pattern for other SDKs?

Reply:
We use a _similar_ pattern in app config with a type guard as shown in the API here: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/appconfiguration/app-configuration/review/app-configuration.api.md?plain=1#L137. I think there are a few different ways to do this; it's an open question I would like to discuss at the arch board meeting.